### PR TITLE
Product page

### PR DESCRIPTION
**Pull request by git-0r**
# Created product page with add to cart functionality
ecom-productpage

## Files to review: 
 - AddToCartButton
 - cartContext
 - pages/Product
 - cartReducer

### CODE REVIEW

File: src/api-calls.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
-// const BASE_URL = "http://localhost:3001";
-// Fake server for deployment
-const BASE_URL = "https://my-json-server.typicode.com/git-0r/fake-server"
+const BASE_URL = "https://ecom-srvr.herokuapp.com";
 
 
 const getProducts = async () => {
     const res = await fetch(`${BASE_URL}/products`);
```
Comment (Ankur9669) [inline src/api-calls.js:5]: Suggestion: Should have tried using axios

File: src/components/AddToCartButton.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    const [cart, updateCart] = useCart();
+    const [productIncart, setProductInCart] = useState(false);
+
+
+    const handleCart = () => {
+        updateCart({ type: "ADD_TO_CART", payload: { ...product, quantity: 1 } })
+    }
+
+    useEffect(() => {
+        const i = cart.findIndex(productIncart => productIncart._id === product?._id);
```
Comment (Ankur9669) [inline src/components/AddToCartButton.js:15]: Suggestion: Variable naming could be better

File: src/components/PopularCategory.js [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+import { getProductsByCategory } from "../api-calls"
 import ProductCard from "./ProductCard"
 
-const PopularCategory = ({ category, products }) => {
+const PopularCategory = ({ category, limit }) => {
+    const [products, setProducts] = useState()
+
+    useEffect(() => {
+        getProductsByCategory(category, limit)
+            .then(data => setProducts(data))
```
Comment (Ankur9669) [inline src/components/PopularCategory.js:10]: fix: Use async await here also because you are using it everywhere

File: src/pages/Product.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { getProductById } from "../api-calls";
+import { useState, useEffect } from "react";
+import AddToCartButton from "../components/AddToCartButton";
+
+const Product = () => {
+  const { id } = useParams();
+  const [product, setProduct] = useState();
+
+  useEffect(() => {
+    getProductById(id).then((data) => setProduct(data));
```
Comment (Ankur9669) [inline src/pages/Product.js:12]: Same use async await

File: src/reducers/cartReducer.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+const cartReducer = (state, action) => {
+    switch (action.type) {
+        case "ADD_TO_CART":
+
+            const newProduct = action.payload;
+
+            const i = state.findIndex(product => product._id === newProduct._id);
```
Comment (Ankur9669) [inline src/reducers/cartReducer.js:7]: Use good variable name

**Review by Ankur9669 — APPROVED** [review]
Most of the things look good, you can look to my suggestions otherwise you can merge it accordig to me

File: src/index.js [hunk 6]
```diff
@@ -2,12 +2,22 @@ import React from "react";
...
 
 ReactDOM.render(
     <React.StrictMode>
         <ProductsProvider>
-            <App />
+            <CartProvider>
+                <BrowserRouter>
+                    <Routes>
+                        <Route path="/" element={<App />} />
+                        <Route path="/product/:id" element={<Product />} />
```
Comment (Naman-Saxena1) [inline src/index.js:17]: I gererally consider App has whole application and then declare routes inside it, not sure if this is good or bad practice.

**Review by Naman-Saxena1 — APPROVED** [review]
Apart from what Ankur suggested, I was not able to find any other issues.
Code looks good to me 👍

**Pull request merged**